Dapper/SqlMapper.TypeHandler.cs

```diff
@@ -16,28 +16,28 @@ public abstract class TypeHandler<T> : ITypeHandler
             /// </summary>
             /// <param name="parameter">The parameter to configure</param>
             /// <param name="value">Parameter value</param>
-            public abstract void SetValue(IDbDataParameter parameter, T value);
+            public abstract void SetValue(IDbDataParameter parameter, T? value);

```

Comment:
@mgravell Since this causes implementers of the class to have to annotate their `T value` parameter as nullable, how would we _handle_ null being passed here? What would it mean?

I know for SqlParameter.Value, `DBNull` means "set to null" and `null` means "unset." Does `null` carry the same "unset" connotation if implementers see it here?

Thanks!

Reply:
Since there isn't a constraint on `T`, this doesn't strictly change the API shape - in particular, a `TypeHandler<SomeValueType>` still has `SetValue(..., SomeValueType)` and not `SetValue(..., SomeValueType?)` ; however, to answer your question: it should mean `DBNull`, but with the opportunity to configure the parameter correctly, and mostly just applies in the `TypeHandler<SomeReferenceType>` scenario, it acknowledges that the incoming parameter *might be* `null`.